### PR TITLE
Add accelerator size to OGCGeometry#estimateMemorySize

### LINKED ISSUE #204: Include the size of geometry accelerator in OGCGeometry#estimateMemorySize
We are seeing acceleration delivering significant performance improvements in spatial joins in Presto:  https://github.com/prestodb/presto/issues/11953 . We'd like to start using this and therefore need to update `OGCGeometry#estimateMemorySize` to account for memory used by the accelerator.

I'll make a PR if there are no objections.

CC: @stolstov

### PR DESCRIPTION
**Pull request by mbasmanova**
Fixes #204

CC: @stolstov @jagill

### CODE REVIEW

File: src/main/java/com/esri/core/geometry/GeometryAccelerators.java [hunk 1]
```diff
@@ -84,4 +82,11 @@ static boolean canUseQuadTreeForPaths(Geometry geom) {
 
         return true;
     }
+
```
Comment (stolstov) [inline src/main/java/com/esri/core/geometry/GeometryAccelerators.java]: @mbasmanova Maria, are these spaces? We use tabs. Check other places, please.

Comment (mbasmanova) [reply src/main/java/com/esri/core/geometry/GeometryAccelerators.java]: @stolstov These must be spaces. I'll fix. Out of curiosity, why do you use tabs?

Comment (stolstov) [reply src/main/java/com/esri/core/geometry/GeometryAccelerators.java]: @mbasmanova Eclipse formats using tabs by default.

Comment (mbasmanova) [reply src/main/java/com/esri/core/geometry/GeometryAccelerators.java]: @stolstov Updated the PR to use tabs.

File: src/main/java/com/esri/core/geometry/QuadTreeImpl.java [hunk 2]
```diff
@@ -777,6 +781,25 @@ QuadTreeSortedIteratorImpl getSortedIterator() {
...
+		long size = SIZE_OF_QUAD_TREE_IMPL +
+				(m_extent != null ? m_extent.estimateMemorySize() : 0) +
+				(m_data_extent != null ? m_data_extent.estimateMemorySize() : 0) +
+				(m_quad_tree_nodes != null ? m_quad_tree_nodes.estimateMemorySize() : 0) +
+				(m_element_nodes != null ? m_element_nodes.estimateMemorySize() : 0) +
+				(m_free_data != null ? m_free_data.estimateMemorySize() : 0);
+
+		if (m_data != null) {
+			size += sizeOfObjectArray(m_data.size());
+			for (Data data : m_data) {
```
Comment (stolstov) [inline src/main/java/com/esri/core/geometry/QuadTreeImpl.java]: Maybe m_data.size() * SIZE_OF_DATA in the for loop?

File: src/test/java/com/esri/core/geometry/TestEstimateMemorySize.java [hunk 3]
```diff
@@ -146,6 +155,38 @@ private void testGeometry(OGCGeometry geometry) {
...
+
+		accelerate(geometry, GeometryAccelerationDegree.enumMedium);
+		long mediumAcceleratedSize = geometry.estimateMemorySize();
+		assertTrue(mediumAcceleratedSize > mildAcceleratedSize);
+
+		accelerate(geometry, GeometryAccelerationDegree.enumHot);
+		long hotAcceleratedSize = geometry.estimateMemorySize();
+		assertTrue(hotAcceleratedSize > mediumAcceleratedSize);
+
+		System.out.println(String.format("%s, %s, %s, %s, %s", initialSize, withEnvelopeSize, mildAcceleratedSize, mediumAcceleratedSize, hotAcceleratedSize));
```
Comment (stolstov) [inline src/test/java/com/esri/core/geometry/TestEstimateMemorySize.java]: Do you want to keep println?

Comment (mbasmanova) [reply src/test/java/com/esri/core/geometry/TestEstimateMemorySize.java]: @stolstov No. I'll remove.

**Comment by mbasmanova** [comment]
@stolstov Sergey, thanks for review. Comments addressed.

**Comment by stolstov** [comment]
@mbasmanova Is this ready to merge?

**Comment by mbasmanova** [comment]
@stolstov Sergey, yes, this is ready to merge unless you have any further comments.

**Comment by stolstov** [comment]
@mbasmanova Maria, thank you for the contribution.

**Pull request merged**